From 42b7861b3233e993458ec5202ed229a0fdf64b09 Mon Sep 17 00:00:00 2001
From: Matthias Vallentin <vallentin@icir.org>
Date: Thu, 9 Jan 2014 16:58:10 -0800
Subject: [PATCH] Disable iossim build to fix linker errors.

When attempting to build the address sanitizer for the iOS simulator, we run
into an architecture mismatch between dependent shared libraries, such as
libc++.dylib.

Disabling the iOS simulator build fixes the issue.
---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a442408..a3e1915 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -188,9 +188,11 @@ if(APPLE)
     OUTPUT_STRIP_TRAILING_WHITESPACE
   )
   set(SANITIZER_COMMON_SUPPORTED_DARWIN_OS osx)
-  if (IOSSIM_SDK_DIR)
-    list(APPEND SANITIZER_COMMON_SUPPORTED_DARWIN_OS iossim)
-  endif()
+  # Disable iOS Simulator build because it clashes with the architecture of
+  # dependent libraries, such libc++abi and libc++.
+  #if (IOSSIM_SDK_DIR)
+  #  list(APPEND SANITIZER_COMMON_SUPPORTED_DARWIN_OS iossim)
+  #endif()
 
   if(COMPILER_RT_USES_LIBCXX)
     set(SANITIZER_MIN_OSX_VERSION 10.7)
-- 
1.8.4.3

